test/runnable/sdtor.d

```diff
@@ -4154,6 +4154,25 @@ int test14815()
 static assert(test14815());
 
 /**********************************/
+// https://issues.dlang.org/show_bug.cgi?id=16197
+
+struct Elem {
+    static string r;
+    int x = -1;
+    this(this) { r ~= 'p'; printf("POSTBLIT %d\n", x++); }
+    ~this()    { r ~= 'd'; printf("DTOR %d\n"    , x++); }
```

Comment:
One unpleasant consequence of this idiom: calling `~=` in a destructor causes reallocation which may cause garbage collection, which aborts the program...